### DISCUSSION TITLE
How to make sure a span with errors can be send to collector without consider the sampler rate?

### DISCUSSION THREAD

**Discussion by lmssy** [open-telemetry/opentelemetry-java-instrumentation#3907] [Q&A]
Hi,

I'm using auto instrumentation to my server and set smapler rate to 0.01, I konw not all spans will be exported to collector.
But I want to make sure all spans with error (like application's exception or those traceId/spanId recorded in error log ) can be send to collector, is there any way to achieve?

**Comment by trask** [comment]
hi @lmssy! The agent and SDK use "head-based" sampling, making their decision at the start of each span, when the error status is not available yet. You may want to look into [tail-sampling provided by the OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/tailsamplingprocessor/README.md). Or alternatively you could possibly implement some kind of tail-sampling in the SpanProcessor pipeline for your exporter depending on your needs.

**Comment by gihad** [comment]
@trask Would this answer change now in 2024? I wanted to configure sampling on the application level and have errors not sample also on the application level, instead of sampling at the collector. Is this possible?

**Reply by trask** [reply]
hi @gihad, I don't believe the above answer has changed
